multik-api/src/jvmMain/kotlin/org/jetbrains/kotlinx/multik/api/Engine.kt

```diff
@@ -0,0 +1,22 @@
+package org.jetbrains.kotlinx.multik.api
+
+import kotlin.reflect.full.createInstance
+
+
+public actual val enginesProvider : Map<EngineType, Engine> get() {
+        val engineClassNames = listOf(
+            "org.jetbrains.kotlinx.multik.jvm.JvmEngine",
+            "org.jetbrains.kotlinx.multik.jni.NativeEngine",
+            "org.jetbrains.kotlinx.multik.default.DefaultEngine")
+        val engines = mutableMapOf<EngineType, Engine>()
+        engineClassNames.forEach { e ->
+            try {
+                val instance = Class.forName(e).kotlin.createInstance() as Engine
+                engines[instance.type] = instance
+            } catch (t: Throwable){ }
+        }
+        return engines
+    }
+
+public actual fun initEnginesProvider(engines: List<Engine>) {
```

Comment:
I am confused by the initialization of engines.
I'll think about it

Reply:
https://github.com/Kotlin/multik/pull/20#issuecomment-998354816